test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -321,6 +325,34 @@ def _create_module():
         for (p1, p2) in zip(fsdp.parameters(), new_fsdp.parameters()):
             self.assertEqual(p1, p2)
 
+    @skip_if_lt_x_gpu(2)
+    def test_state_dict_with_ignored_modules(self):
+        # Initialize an FSDP-wrapped model with an ignored module
+        model = Model(wrap_fsdp=True).cuda()
+        ignored_modules = [model.outer]
+        ignored_param_to_param_name = {
+            model.outer.bias: "outer.bias", model.outer.weight: "outer.weight",
+        }
+        fsdp_model = FSDP(model, ignored_modules=ignored_modules)
```

Comment:
if `ignored_modules` is part of `model` that is already wrapped in FSDP, will they be ignored? Seems unlikely as these modules were probably already sharded by the first call (i.e. L331) and we can't "undo" that?